Author: <bdefreese@bdubuntu3>
Description: Fix g++ colorizing issue
--- a/colorgcc
+++ b/colorgcc
@@ -170,7 +170,7 @@
    # This substitute replaces `foo' with `AfooB' where A is the escape
    # sequence that turns on the the desired source color, and B is the
    # escape sequence that returns to $normalColor.
-   $line =~ s/\`(.*?)\'/\`$srcon$1$srcoff\'/g;
+   $line =~ s/(\`|\')(.*?)\'/\`$srcon$2$srcoff\'/g;
 
    print($line, color("reset"));
 }
